carma_ros2_utils/src/lifecycle_component_wrapper.cpp

```diff
@@ -0,0 +1,410 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Modifications copyright (C) 2021 Leidos
+ * - Converted into Lifecycle Component Wrapper
+ * 
+ */ 
+
+#include "carma_ros2_utils/lifecycle_component_wrapper.hpp"
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <ament_index_cpp/get_resource.hpp>
+#include <class_loader/class_loader.hpp>
+#include <rcpputils/filesystem_helper.hpp>
+#include <rcpputils/split.hpp>
+
+using namespace std::placeholders;
+
+namespace carma_ros2_utils
+{
+
+///// CARMA CHANGE /////
+LifecycleComponentWrapper::LifecycleComponentWrapper(const rclcpp::NodeOptions & node_options)
+: carma_ros2_utils::CarmaLifecycleNode(node_options)
+{
+  // Do nothing
+}
+
+void LifecycleComponentWrapper::initialize(std::weak_ptr<rclcpp::Executor> executor) {
+  executor_ = executor;
+
+  loadNode_srv_ = create_service<LoadNode>(
+    "~/_container/load_node",
```

Comment:
is the leading `_` to imply private, similar to how it's used in python? Or is this formally enforced in ROS2?

Reply:
The ~ indicates private. I image the '_' is to avoid naming conflicts. This is the formal interface for the launch file component managers.